### ISSUE TITLE
jepsen/store/FressianReader.java: [unchecked] compiler warnings (new w/latest commits)

### ISSUE THREAD

**Issue by nurturenature** [jepsen-io/jepsen#557]
Hi,

Pulling the latest commits and doing a local 0.2.8-SNAPSHOT build/install generates new compiler warnings.
Previously there were none.

Enabling `-Xlint:unchecked` shows:

```
.../jepsen-io/jepsen/jepsen/src/jepsen/store/FressianReader.java:675: warning: [unchecked] unchecked call to add(E) as a member of the raw type ArrayList
                getStructCache().add(new StructType(tag, fields));
                                    ^
  where E is a type-variable:
    E extends Object declared in class ArrayList
/home/jsuttor/projects/jepsen-io/jepsen/jepsen/src/jepsen/store/FressianReader.java:772: warning: [unchecked] unchecked conversion
        ArrayList<byte[]> chunks = new ArrayList();
                                   ^
  required: ArrayList<byte[]>
  found:    ArrayList

...

12 warnings
```

Tests run fine with latest commits.

**Comment by aphyr** [comment]
If you want to patch this feel free! It's probably the same in the upstream fressian package too, but we're not seeing it because it's normally compiled by the time the jar gets to us.

**Comment by nurturenature** [comment]
Closing as fixed in #558.

Thanks!

(P.S. Do have the last several commits up in tabs. 🙂   Most looking forward to streaming histories. A partial history on crash makes more tests good tests, i.e. provide insight. I've also had to dial some tests back due to client crashes. And now I can test the crash. Second is being able to point analyze directly at a test run.)
